// @hongbeccodeduocchua/fo4-db/dist/utils/enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Regions = exports.PreferFoot = exports.WorkRate = exports.BodyType = exports.TeamColor = exports.Countries = exports.HiddenStats = exports.sPositions = exports.Attributes = exports.Seasons = exports.Positions = void 0;
var Positions;
(function (Positions) {
    Positions["LW"] = "lw";
    Positions["ST"] = "st";
    Positions["RW"] = "rw";
    Positions["CF"] = "cf";
    Positions["CAM"] = "cam";
    Positions["RM"] = "rm";
    Positions["LM"] = "lm";
    Positions["CM"] = "cm";
    Positions["CDM"] = "cdm";
    Positions["LWB"] = "lwb";
    Positions["RWB"] = "rwb";
    Positions["LB"] = "lb";
    Positions["RB"] = "rb";
    Positions["CB"] = "cb";
    Positions["GK"] = "gk";
})(Positions = exports.Positions || (exports.Positions = {}));
var Seasons;
(function (Seasons) {
    Seasons["ICON"] = "ICON";
    Seasons["BWC"] = "BWC";
    Seasons["WC22"] = "WC22";
    Seasons["23TY"] = "23TY";
    Seasons["23TYN"] = "23TYN";
    Seasons["22TY"] = "22TY";
    Seasons["22TYN"] = "22TYN";
    Seasons["21TY"] = "21TY";
    Seasons["21TYN"] = "21TYN";
    Seasons["20TY"] = "20TY";
    Seasons["20TYN"] = "20TYN";
    Seasons["19TY"] = "19TY";
    Seasons["18TY"] = "18TY";
    Seasons["22TS"] = "22TS";
    Seasons["21TS"] = "21TS";
    Seasons["20TS"] = "20TS";
    Seasons["19TS"] = "19TS";
    Seasons["NHD"] = "NHD";
    Seasons["TB"] = "TB";
    Seasons["TT"] = "TT";
    Seasons["GR"] = "GR";
    Seasons["TC"] = "TC";
    Seasons["HOT"] = "HOT";
    Seasons["SPL"] = "SPL";
    Seasons["LN"] = "LN";
    Seasons["LOL"] = "LOL";
    Seasons["FA"] = "FA";
    Seasons["OTW"] = "OTW";
    Seasons["BOE"] = "BOE";
    Seasons["BOE21"] = "BOE21";
    Seasons["BTB"] = "BTB";
    Seasons["CAP"] = "CAP";
    Seasons["EBS"] = "EBS";
    Seasons["UP"] = "UP";
    Seasons["MC"] = "MC";
    Seasons["VTR"] = "VTR";
    Seasons["MOG"] = "MOG";
    Seasons["LH"] = "LH";
    Seasons["COC"] = "COC";
    Seasons["22HR"] = "22HR";
    Seasons["22NG"] = "22NG";
    Seasons["21NG"] = "21NG";
    Seasons["20NG"] = "20NG";
    Seasons["19NG"] = "19NG";
    Seasons["21UCL"] = "21UCL";
    Seasons["20UCL"] = "20UCL";
    Seasons["19UCL"] = "19UCL";
    Seasons["22PL"] = "22PL";
    Seasons["21PL"] = "21PL";
    Seasons["20A"] = "20A";
    Seasons["19S"] = "19S";
    Seasons["19A"] = "19A";
    Seasons["18S"] = "18S";
    Seasons["18A"] = "18A";
    Seasons["LIVE"] = "LIVE";
    Seasons["21Live"] = "21";
    Seasons["20Live"] = "20";
    Seasons["19Live"] = "19";
    Seasons["18Live"] = "18";
    Seasons["17Live"] = "17";
    Seasons["NTG"] = "NTG";
    Seasons["19WL"] = "19WL";
    Seasons["RMFC"] = "RMFC";
    Seasons["CFA"] = "CFA";
    Seasons["LA"] = "LA";
    Seasons["PSG"] = "PSG";
    Seasons["MCI"] = "MCI";
    Seasons["MCC"] = "MCC";
    Seasons["THL"] = "THL";
    Seasons["VN"] = "VN";
    Seasons["TKI"] = "TKI";
    Seasons["TKL"] = "TKL";
    Seasons["20KL"] = "20KL";
    Seasons["20KLB"] = "20KLB";
    Seasons["K19"] = "K19";
    Seasons["K18"] = "K18";
})(Seasons = exports.Seasons || (exports.Seasons = {}));
var Attributes;
(function (Attributes) {
    Attributes["SPRINTSPEED"] = "sprintspeed";
    Attributes["ACCELERATION"] = "acceleration";
    Attributes["FINISHING"] = "finishing";
    Attributes["SHOTPOWER"] = "shotpower";
    Attributes["LONGSHOTS"] = "longshots";
    Attributes["POSITIONING"] = "positioning";
    Attributes["VOLLEYS"] = "volleys";
    Attributes["PENALTIES"] = "penalties";
    Attributes["SHORTPASSING"] = "shortpassing";
    Attributes["VISION"] = "vision";
    Attributes["CROSSING"] = "crossing";
    Attributes["LONGPASSING"] = "longpassing";
    Attributes["FREEKICKACCURACY"] = "freekickaccuracy";
    Attributes["CURVE"] = "curve";
    Attributes["DRIBBLING"] = "dribbling";
    Attributes["BALLCONTROL"] = "ballcontrol";
    Attributes["AGILITY"] = "agility";
    Attributes["BALANCE"] = "balance";
    Attributes["REACTIONS"] = "reactions";
    Attributes["MARKING"] = "marking";
    Attributes["STANDINGTACKLE"] = "standingtackle";
    Attributes["INTERCEPTIONS"] = "interceptions";
    Attributes["HEADINGACCURACY"] = "headingaccuracy";
    Attributes["SLIDINGTACKLE"] = "slidingtackle";
    Attributes["STRENGTH"] = "strength";
    Attributes["STAMINA"] = "stamina";
    Attributes["AGGRESSION"] = "aggression";
    Attributes["JUMPING"] = "jumping";
    Attributes["COMPOSURE"] = "composure";
    Attributes["GKDIVING"] = "gkdiving";
    Attributes["GKHANDLING"] = "gkhandling";
    Attributes["GKKICKING"] = "gkkicking";
    Attributes["GKREFLEXES"] = "gkreflexes";
    Attributes["GKPOSITIONING"] = "gkpositioning";
})(Attributes = exports.Attributes || (exports.Attributes = {}));
var sPositions;
(function (sPositions) {
    sPositions["GK"] = "gk";
    sPositions["CB"] = "cb";
    sPositions["RB"] = "rb";
    sPositions["RWB"] = "rwb";
    sPositions["CDM"] = "cdm";
    sPositions["CM"] = "cm";
    sPositions["RM"] = "rm";
    sPositions["CAM"] = "cam";
    sPositions["RW"] = "rw";
    sPositions["CF"] = "cf";
    sPositions["ST"] = "st";
})(sPositions = exports.sPositions || (exports.sPositions = {}));
var HiddenStats;
(function (HiddenStats) {
    HiddenStats["acrobatic-clearance"] = "acrobatic-clearance";
    HiddenStats["argues-with-officials"] = "argues-with-officials";
    HiddenStats["avoids-using-weaker-foot"] = "avoids-using-weaker-foot";
    HiddenStats["backs-into-player"] = "backs-into-player";
    HiddenStats["corner-specialist"] = "corner-specialist";
    HiddenStats["diver"] = "diver";
    HiddenStats["dives-into-tackles"] = "dives-into-tackles";
    HiddenStats["early-crosser"] = "early-crosser";
    HiddenStats["finesse-free-kick"] = "finesse-free-kick";
    HiddenStats["finesse-shot"] = "finesse-shot";
    HiddenStats["flair"] = "flair";
    HiddenStats["giant-throw-in"] = "giant-throw-in";
    HiddenStats["gk-cautious-with-crosses"] = "gk-cautious-with-crosses";
    HiddenStats["gk-comes-for-crosses"] = "gk-comes-for-crosses";
    HiddenStats["gk-long-thrower"] = "gk-long-thrower";
    HiddenStats["gk-one-on-ones"] = "gk-one-on-ones";
    HiddenStats["inflexibility"] = "inflexibility";
    HiddenStats["injury-prone"] = "injury-prone";
    HiddenStats["leadership"] = "leadership";
    HiddenStats["long-passer"] = "long-passer";
    HiddenStats["long-shot-taker"] = "long-shot-taker";
    HiddenStats["long-throw-in"] = "long-throw-in";
    HiddenStats["one-club-player"] = "one-club-player";
    HiddenStats["outside-foot-shot"] = "outside-foot-shot";
    HiddenStats["play-maker"] = "play-maker";
    HiddenStats["power-header"] = "power-header";
    HiddenStats["puncher"] = "puncher";
    HiddenStats["pushes-up-for-corners"] = "pushes-up-for-corners";
    HiddenStats["second-wind"] = "second-wind";
    HiddenStats["selfish"] = "selfish";
    HiddenStats["shooting-chip-shot"] = "shooting-chip-shot";
    HiddenStats["solid-player"] = "solid-player";
    HiddenStats["speed-dribbler"] = "speed-dribbler";
    HiddenStats["sweeper-keeper"] = "sweeper-keeper";
    HiddenStats["swerve"] = "swerve";
    HiddenStats["takes-powerful-driven-free-kicks"] = "takes-powerful-driven-free-kicks";
    HiddenStats["target-forward"] = "target-forward";
    HiddenStats["team-player"] = "team-player";
    HiddenStats["technical-dribbler"] = "technical-dribbler";
    HiddenStats["tries-to-beat-offside-trap"] = "tries-to-beat-offside-trap";
})(HiddenStats = exports.HiddenStats || (exports.HiddenStats = {}));
var Countries;
(function (Countries) {
    Countries["afghanistan"] = "afghanistan";
    Countries["albania"] = "albania";
    Countries["algeria"] = "algeria";
    Countries["american-samoa"] = "american-samoa";
    Countries["andorra"] = "andorra";
    Countries["angola"] = "angola";
    Countries["anguilla"] = "anguilla";
    Countries["antigua-and-barbuda"] = "antigua-and-barbuda";
    Countries["argentina"] = "argentina";
    Countries["armenia"] = "armenia";
    Countries["aruba"] = "aruba";
    Countries["australia"] = "australia";
    Countries["austria"] = "austria";
    Countries["azerbaijan"] = "azerbaijan";
    Countries["bahamas"] = "bahamas";
    Countries["bahrain"] = "bahrain";
    Countries["bangladesh"] = "bangladesh";
    Countries["barbados"] = "barbados";
    Countries["belarus"] = "belarus";
    Countries["belgium"] = "belgium";
    Countries["belize"] = "belize";
    Countries["benin"] = "benin";
    Countries["bermuda"] = "bermuda";
    Countries["bhutan"] = "bhutan";
    Countries["bolivia"] = "bolivia";
    Countries["bosnia-herzegovina"] = "bosnia-herzegovina";
    Countries["botswana"] = "botswana";
    Countries["brazil"] = "brazil";
    Countries["british-virgin-is"] = "british-virgin-is";
    Countries["brunei-darussalam"] = "brunei-darussalam";
    Countries["bulgaria"] = "bulgaria";
    Countries["burkina-faso"] = "burkina-faso";
    Countries["burundi"] = "burundi";
    Countries["cambodia"] = "cambodia";
    Countries["cameroon"] = "cameroon";
    Countries["canada"] = "canada";
    Countries["cape-verde-islands"] = "cape-verde-islands";
    Countries["cayman-islands"] = "cayman-islands";
    Countries["central-african-rep"] = "central-african-rep";
    Countries["chad"] = "chad";
    Countries["chile"] = "chile";
    Countries["china-pr"] = "china-pr";
    Countries["chinese-taipei"] = "chinese-taipei";
    Countries["colombia"] = "colombia";
    Countries["comoros"] = "comoros";
    Countries["congo"] = "congo";
    Countries["cook-islands"] = "cook-islands";
    Countries["costa-rica"] = "costa-rica";
    Countries["created-players-country"] = "created-players-country";
    Countries["creation-zone"] = "creation-zone";
    Countries["croatia"] = "croatia";
    Countries["cuba"] = "cuba";
    Countries["cura\u00E7ao"] = "cura\u00E7ao";
    Countries["cyprus"] = "cyprus";
    Countries["czech-republic"] = "czech-republic";
    Countries["denmark"] = "denmark";
    Countries["djibouti"] = "djibouti";
    Countries["dominica"] = "dominica";
    Countries["dominican-republic"] = "dominican-republic";
    Countries["dr-congo"] = "dr-congo";
    Countries["ecuador"] = "ecuador";
    Countries["egypt"] = "egypt";
    Countries["el-salvador"] = "el-salvador";
    Countries["england"] = "england";
    Countries["equatorial-guinea"] = "equatorial-guinea";
    Countries["eritrea"] = "eritrea";
    Countries["estonia"] = "estonia";
    Countries["ethiopia"] = "ethiopia";
    Countries["faroe-islands"] = "faroe-islands";
    Countries["fiji"] = "fiji";
    Countries["finland"] = "finland";
    Countries["france"] = "france";
    Countries["free-agents-country"] = "free-agents-country";
    Countries["fyr-macedonia"] = "fyr-macedonia";
    Countries["gabon"] = "gabon";
    Countries["gambia"] = "gambia";
    Countries["georgia"] = "georgia";
    Countries["germany"] = "germany";
    Countries["ghana"] = "ghana";
    Countries["gibraltar"] = "gibraltar";
    Countries["greece"] = "greece";
    Countries["greenland"] = "greenland";
    Countries["grenada"] = "grenada";
    Countries["guam"] = "guam";
    Countries["guatemala"] = "guatemala";
    Countries["guinea"] = "guinea";
    Countries["guinea-bissau"] = "guinea-bissau";
    Countries["guyana"] = "guyana";
    Countries["haiti"] = "haiti";
    Countries["holland"] = "holland";
    Countries["honduras"] = "honduras";
    Countries["hong-kong"] = "hong-kong";
    Countries["hungary"] = "hungary";
    Countries["iceland"] = "iceland";
    Countries["india"] = "india";
    Countries["indonesia"] = "indonesia";
    Countries["international"] = "international";
    Countries["iran"] = "iran";
    Countries["iraq"] = "iraq";
    Countries["ireland-republic"] = "ireland-republic";
    Countries["israel"] = "israel";
    Countries["italy"] = "italy";
    Countries["ivory-coast"] = "ivory-coast";
    Countries["jamaica"] = "jamaica";
    Countries["japan"] = "japan";
    Countries["jordan"] = "jordan";
    Countries["kazakhstan"] = "kazakhstan";
    Countries["kenya"] = "kenya";
    Countries["korea-dpr"] = "korea-dpr";
    Countries["korea-republic"] = "korea-republic";
    Countries["kuwait"] = "kuwait";
    Countries["kyrgyzstan"] = "kyrgyzstan";
    Countries["laos"] = "laos";
    Countries["latvia"] = "latvia";
    Countries["lebanon"] = "lebanon";
    Countries["lesotho"] = "lesotho";
    Countries["liberia"] = "liberia";
    Countries["libya"] = "libya";
    Countries["liechtenstein"] = "liechtenstein";
    Countries["lithuania"] = "lithuania";
    Countries["luxembourg"] = "luxembourg";
    Countries["macau"] = "macau";
    Countries["madagascar"] = "madagascar";
    Countries["malawi"] = "malawi";
    Countries["malaysia"] = "malaysia";
    Countries["maldives"] = "maldives";
    Countries["mali"] = "mali";
    Countries["malta"] = "malta";
    Countries["mauritania"] = "mauritania";
    Countries["mauritius"] = "mauritius";
    Countries["mexico"] = "mexico";
    Countries["moldova"] = "moldova";
    Countries["mongolia"] = "mongolia";
    Countries["montenegro"] = "montenegro";
    Countries["montserrat"] = "montserrat";
    Countries["morocco"] = "morocco";
    Countries["mozambique"] = "mozambique";
    Countries["myanmar"] = "myanmar";
    Countries["namibia"] = "namibia";
    Countries["nepal"] = "nepal";
    Countries["new-caledonia"] = "new-caledonia";
    Countries["new-zealand"] = "new-zealand";
    Countries["nicaragua"] = "nicaragua";
    Countries["niger"] = "niger";
    Countries["nigeria"] = "nigeria";
    Countries["northern-ireland"] = "northern-ireland";
    Countries["norway"] = "norway";
    Countries["oman"] = "oman";
    Countries["pakistan"] = "pakistan";
    Countries["palestinian-authority"] = "palestinian-authority";
    Countries["panama"] = "panama";
    Countries["papua-new-guinea"] = "papua-new-guinea";
    Countries["paraguay"] = "paraguay";
    Countries["peru"] = "peru";
    Countries["philippines"] = "philippines";
    Countries["poland"] = "poland";
    Countries["portugal"] = "portugal";
    Countries["puerto-rico"] = "puerto-rico";
    Countries["qatar"] = "qatar";
    Countries["rest-of-world"] = "rest-of-world";
    Countries["romania"] = "romania";
    Countries["russia"] = "russia";
    Countries["rwanda"] = "rwanda";
    Countries["samoa"] = "samoa";
    Countries["san-marino"] = "san-marino";
    Countries["s\u00E3o-tom\u00E9-and-pr\u00EDncipe"] = "s\u00E3o-tom\u00E9-and-pr\u00EDncipe";
    Countries["saudi-arabia"] = "saudi-arabia";
    Countries["scotland"] = "scotland";
    Countries["senegal"] = "senegal";
    Countries["serbia"] = "serbia";
    Countries["seychelles"] = "seychelles";
    Countries["sierra-leone"] = "sierra-leone";
    Countries["singapore"] = "singapore";
    Countries["slovakia"] = "slovakia";
    Countries["slovenia"] = "slovenia";
    Countries["solomon-islands"] = "solomon-islands";
    Countries["somalia"] = "somalia";
    Countries["south-africa"] = "south-africa";
    Countries["south-sudan"] = "south-sudan";
    Countries["spain"] = "spain";
    Countries["sri-lanka"] = "sri-lanka";
    Countries["st-kitts-nevis"] = "st-kitts-nevis";
    Countries["st-lucia"] = "st-lucia";
    Countries["st-vincent-and-the-grenadines"] = "st-vincent-and-the-grenadines";
    Countries["sudan"] = "sudan";
    Countries["suriname"] = "suriname";
    Countries["swaziland"] = "swaziland";
    Countries["sweden"] = "sweden";
    Countries["switzerland"] = "switzerland";
    Countries["syria"] = "syria";
    Countries["tahiti"] = "tahiti";
    Countries["tajikistan"] = "tajikistan";
    Countries["tanzania"] = "tanzania";
    Countries["thailand"] = "thailand";
    Countries["timor-leste"] = "timor-leste";
    Countries["togo"] = "togo";
    Countries["tonga"] = "tonga";
    Countries["trinidad-and-tobago"] = "trinidad-and-tobago";
    Countries["tunisia"] = "tunisia";
    Countries["turkey"] = "turkey";
    Countries["turkmenistan"] = "turkmenistan";
    Countries["turks-and-caicos-islands"] = "turks-and-caicos-islands";
    Countries["uganda"] = "uganda";
    Countries["ukraine"] = "ukraine";
    Countries["united-arab-emirates"] = "united-arab-emirates";
    Countries["united-states"] = "united-states";
    Countries["uruguay"] = "uruguay";
    Countries["us-virgin-islands"] = "us-virgin-islands";
    Countries["uzbekistan"] = "uzbekistan";
    Countries["vanuatu"] = "vanuatu";
    Countries["venezuela"] = "venezuela";
    Countries["vietnam"] = "vietnam";
    Countries["wales"] = "wales";
    Countries["yemen"] = "yemen";
    Countries["zambia"] = "zambia";
    Countries["zimbabwe"] = "zimbabwe";
})(Countries = exports.Countries || (exports.Countries = {}));
var TeamColor;
(function (TeamColor) {
    TeamColor["manchester-city"] = "manchester-city";
    TeamColor["liverpool"] = "liverpool";
    TeamColor["tottenham-hotspur"] = "tottenham-hotspur";
    TeamColor["chelsea"] = "chelsea";
    TeamColor["arsenal"] = "arsenal";
    TeamColor["manchester-united"] = "manchester-united";
    TeamColor["wolverhampton-wanderers"] = "wolverhampton-wanderers";
    TeamColor["leicester-city"] = "leicester-city";
    TeamColor["everton"] = "everton";
    TeamColor["watford"] = "watford";
    TeamColor["fc-bayern-munchen"] = "fc-bayern-munchen";
    TeamColor["borussia-dortmund"] = "borussia-dortmund";
    TeamColor["rb-leipzig"] = "rb-leipzig";
    TeamColor["eintracht-frankfurt"] = "eintracht-frankfurt";
    TeamColor["bayer-04-leverkusen"] = "bayer-04-leverkusen";
    TeamColor["vfl-wolfsburg"] = "vfl-wolfsburg";
    TeamColor["fc-barcelona"] = "fc-barcelona";
    TeamColor["atletico-de-madrid"] = "atletico-de-madrid";
    TeamColor["real-madrid"] = "real-madrid";
    TeamColor["valencia-cf"] = "valencia-cf";
    TeamColor["rcd-espanyol-de-barcelona"] = "rcd-espanyol-de-barcelona";
    TeamColor["piemonte-calcio"] = "piemonte-calcio";
    TeamColor["napoli"] = "napoli";
    TeamColor["inter"] = "inter";
    TeamColor["roma-fc"] = "roma-fc";
    TeamColor["milan"] = "milan";
    TeamColor["lazio"] = "lazio";
    TeamColor["paris-saint-germain"] = "paris-saint-germain";
    TeamColor["losc-lille"] = "losc-lille";
    TeamColor["olympique-lyonnais"] = "olympique-lyonnais";
    TeamColor["olympique-de-marseille"] = "olympique-de-marseille";
    TeamColor["holland-eredivisie"] = "holland-eredivisie";
    TeamColor["ajax"] = "ajax";
    TeamColor["psv"] = "psv";
    TeamColor["jeonbuk-hyundai-motors"] = "jeonbuk-hyundai-motors";
    TeamColor["ulsan-hyundai"] = "ulsan-hyundai";
    TeamColor["fc-seoul"] = "fc-seoul";
    TeamColor["daegu-fc"] = "daegu-fc";
    TeamColor["sangju-sangmu-fc"] = "sangju-sangmu-fc";
    TeamColor["seongnam-fc"] = "seongnam-fc";
    TeamColor["gangwon-fc"] = "gangwon-fc";
    TeamColor["pohang-steelers"] = "pohang-steelers";
    TeamColor["gyeongnam-fc"] = "gyeongnam-fc";
    TeamColor["suwon-samsung-bluewings"] = "suwon-samsung-bluewings";
    TeamColor["incheon-united-fc"] = "incheon-united-fc";
    TeamColor["jeju-united-fc"] = "jeju-united-fc";
    TeamColor["guangzhou-fc"] = "guangzhou-fc";
    TeamColor["shanghai-port-fc"] = "shanghai-port-fc";
    TeamColor["jiangsu-suning"] = "jiangsu-suning";
    TeamColor["shandong-taishan"] = "shandong-taishan";
    TeamColor["chongqing-liangjiang-athletic"] = "chongqing-liangjiang-athletic";
    TeamColor["tianjin-jinmen-tiger-fc"] = "tianjin-jinmen-tiger-fc";
    TeamColor["guangzhou-city"] = "guangzhou-city";
    TeamColor["shanghai-shenhua"] = "shanghai-shenhua";
})(TeamColor = exports.TeamColor || (exports.TeamColor = {}));
var BodyType;
(function (BodyType) {
    BodyType["LARGE"] = "large";
    BodyType["NORMAL"] = "normal";
    BodyType["SMALL"] = "small";
})(BodyType = exports.BodyType || (exports.BodyType = {}));
var WorkRate;
(function (WorkRate) {
    WorkRate["HIGH"] = "high";
    WorkRate["LOW"] = "low";
    WorkRate["MID"] = "mid";
})(WorkRate = exports.WorkRate || (exports.WorkRate = {}));
var PreferFoot;
(function (PreferFoot) {
    PreferFoot["LEFT"] = "left";
    PreferFoot["RIGHT"] = "right";
})(PreferFoot = exports.PreferFoot || (exports.PreferFoot = {}));
var Regions;
(function (Regions) {
    Regions["VN"] = "vn";
    Regions["KR"] = "kr";
    Regions["CN"] = "cn";
    Regions["ID"] = "id";
    Regions["RU"] = "ru";
    Regions["EN"] = "en";
    Regions["TH"] = "th";
})(Regions = exports.Regions || (exports.Regions = {}));
